src/strategy/flexible-rollout-strategy.ts

```diff
@@ -19,16 +17,14 @@ export class FlexibleRolloutStrategy extends Strategy {
         }
     }
 
-    resolveStickiness(stickiness: string, context: Context): any {
+    resolveStickiness(stickiness: string | undefined, context: Context): any {
         switch (stickiness) {
-            case STICKINESS.userId:
-                return context.userId;
-            case STICKINESS.sessionId:
-                return context.sessionId;
+            case STICKINESS.default:
+                return context.userId || context.sessionId || this.randomGenerator();
             case STICKINESS.random:
                 return this.randomGenerator();
             default:
-                return context.userId || context.sessionId || this.randomGenerator();
+                return stickiness ? context[stickiness] : null;
```

Comment:
Hi, This almost correct. Because a poor decision in the early days, pre typescript, the context is a bit ambiguous when it comes to custom context fields. The initial idea was that all "not official defined context fields" should be defined under `context.properties` ([source code](https://github.com/Unleash/unleash-client-node/blob/master/src/context.ts#L12)). 

This means that we need to first check "context[stickiness]" and if it is undefined we also need to look inside "context.properties[stickiness]" (and context.properties may also be undefined).

Reply:
There is actually a helper for this:
https://github.com/Unleash/unleash-client-node/blob/master/src/helpers.ts#L19